test/distributed/_shard/sharded_tensor/test_sharded_tensor.py

```diff
@@ -1781,6 +1785,124 @@ def test_with_rpc_names(self):
                 self.assertEqual((5, 5), shard.tensor.size())
 
 
+class TestShardedTensorFromLocalTensor(ShardedTensorTestBase):
+    def _generate_st_from_chunk_local_tensor(self, st_size, sharding_spec):
+        chunk_sharding_dim = sharding_spec.dim
+        world_size = len(sharding_spec.placements)
+        split_size = get_split_size(st_size[chunk_sharding_dim], world_size)
+        shards_metadata = [None] * world_size
+        local_shard_meta = None
+        for idx, placement in enumerate(sharding_spec.placements):
+            shard_size = copy.deepcopy(st_size)
+            offsets = [0] * len(st_size)
+            offsets[chunk_sharding_dim] = min(
```

Comment:
I guess we will have a easier way to generate the ShardedMetadatas for sharding spec when we have https://github.com/pytorch/pytorch/pull/72130

Reply:
Yes, that's right. I will send a PR to rebase on your change. Let me add a TODO here.